<a href="https://colab.research.google.com/github/Alb3rtoGitHub/utilizando-chatgpt-como-asistente/blob/main/Utilizando_chatgpt_como_asistente.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <strong><font color="7b71ff">Análisis de datos con python: Utilizando ChatGPT como asistente</font></strong>


---

<img src="https://raw.githubusercontent.com/alura-cursos/ia-datascience-dataviz/main/imagens/logos/logo_branca_fundo_azul.png" alt="inserir alt">

# **Aula 1 - Conociendo los datos**

 Zoop es una gran minorista que atiende a todas las regiones de México a través de su e-commerce. Es conocida por la amplia variedad de productos que ofrece, buscando satisfacer a todo tipo de público. Posee una serie de centros de distribución (CDs) para la entrega de sus productos, la mayoría de ellos situados en el estado de Mexico. También posee una cadena de supermercados, y vamos a trabajar como científicos de datos para esta prestigiosa empresa.

### **Accesando a los datos**

In [2]:
# url = 'https://gist.githubusercontent.com/ahcamachod/24a6235f7cdf255501b5ab4f9b0fd97f/raw/fa329f6d6c1b6b00b3ba457cdc850b53210633ca/zoop_market.json'
url = 'https://gist.githubusercontent.com/ahcamachod/4b964953032a2078d67052ec8fb92e72/raw/8106d7c3f001390744ec3aa657b9847f306c9df8/zoop_supermercado.json' # sacado de Preparando el ambiente

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>

Vamos a actuar como científicos de datos en Zoop.

Recibimos datos en formato de archivo JSON, estructurados en listas y diccionarios. El archivo está disponible en una URL de Github.

Necesitamos analizar esos datos para ayudar en las decisiones de la empresa. Programamos en Python y conocemos las bibliotecas de análisis de datos Numpy y Pandas. Por lo tanto, prioriza el uso de estas bibliotecas.

Igualmente, infórmanos cómo podemos leer estos datos en formato JSON y para poder visualizarlos en formato de tabla en Python.

In [3]:
import pandas as pd
import numpy as np

#Leer el archivo JSON desde la url
df = pd.read_json(url)

In [4]:
df

item_identificador tienda_identificador  ventas_totales  \
0                 FDB08               OUT018       176503.58   
1                 DRQ35               OUT049       185758.20   
2                 FDD14               OUT018       165983.94   
3                 FDY37               OUT045       314923.40   
4                 FDY59               OUT018        64782.34   
...                 ...                  ...             ...   
8545              FDY08               OUT010        28096.76   
8546              FDC41               OUT017       130163.90   
8547              NCQ53               OUT045       614533.40   
8548              FDL46               OUT017       164985.24   
8549              NCN30               OUT046        96541.00   

                                                   item  \
0     {'item_peso': 6.055, 'item_contenido_grasa': '...   
1     {'item_peso': 9.3, 'item_contenido_grasa': 'Ba...   
2     {'item_peso': 20.7, 'item_contenido_grasa': 'B...   
3     {'item_peso': 17.0, 'item_contenido_grasa': 'R...   
4     {'item_peso': 8.195, 'item_contenido_grasa': '...   
...                                                 ...   
8545  {'item_peso': 9.395, 'item_contenido_grasa': '...   
8546  {'item_peso': 15.6, 'item_contenido_grasa': 'B...   
8547  {'item_peso': 17.6, 'item_contenido_grasa': 'B...   
8548  {'item_peso': 20.35, 'item_contenido_grasa': '...   
8549  {'item_peso': 16.35, 'item_contenido_grasa': '...   

                                                 tienda  
0     {'tienda_ano_establecimiento': 2019, 'tienda_t...  
1     {'tienda_ano_establecimiento': 2009, 'tienda_t...  
2     {'tienda_ano_establecimiento': 2019, 'tienda_t...  
3     {'tienda_ano_establecimiento': 2012, 'tienda_t...  
4     {'tienda_ano_establecimiento': 2019, 'tienda_t...  
...                                                 ...  
8545  {'tienda_ano_establecimiento': 2008, 'tienda_t...  
8546  {'tienda_ano_establecimiento': 2017, 'tienda_t...  
8547  {'tienda_ano_establecimiento': 2012, 'tienda_t...  
8548  {'tienda_ano_establecimiento': 2017, 'tienda_t...  
8549  {'tienda_ano_establecimiento': 2007, 'tienda_t...  

[8550 rows x 5 columns]

In [5]:
# Mostrar las primeras filas
print(df.head())

# Ver información general del DataFrame
print(df.info())

# Estadísticas descriptivas (solo para columnas numéricas)
print(df.describe())

  item_identificador tienda_identificador  ventas_totales  \
0              FDB08               OUT018       176503.58   
1              DRQ35               OUT049       185758.20   
2              FDD14               OUT018       165983.94   
3              FDY37               OUT045       314923.40   
4              FDY59               OUT018        64782.34   

                                                item  \
0  {'item_peso': 6.055, 'item_contenido_grasa': '...   
1  {'item_peso': 9.3, 'item_contenido_grasa': 'Ba...   
2  {'item_peso': 20.7, 'item_contenido_grasa': 'B...   
3  {'item_peso': 17.0, 'item_contenido_grasa': 'R...   
4  {'item_peso': 8.195, 'item_contenido_grasa': '...   

                                              tienda  
0  {'tienda_ano_establecimiento': 2019, 'tienda_t...  
1  {'tienda_ano_establecimiento': 2009, 'tienda_t...  
2  {'tienda_ano_establecimiento': 2019, 'tienda_t...  
3  {'tienda_ano_establecimiento': 2012, 'tienda_t...  
4  {'tienda_ano_estab

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>


El DataFrame "df" posee las columnas "item_identificador", "tienda_identificador", "ventas_totales", "item" y "tienda".

Las columnas "item" y "tienda" contienen diccionarios anidados dentro de cada una de las filas. ¿Cómo puedo transformar los datos de los diccionarios y columnas para hacerlos más accesibles en mi DataFrame?


In [6]:
df_items = pd.json_normalize(df["item"])
df_tiendas = pd.json_normalize(df["tienda"])

In [7]:
# Concatenar los nuevos DataFrames con el original (sin las columnas anidadas)
df = pd.concat([df.drop(columns=["item", "tienda"], axis=1), df_items, df_tiendas], axis=1)

In [8]:
df.head()

item_identificador tienda_identificador  ventas_totales  item_peso  \
0              FDB08               OUT018       176503.58      6.055   
1              DRQ35               OUT049       185758.20      9.300   
2              FDD14               OUT018       165983.94     20.700   
3              FDY37               OUT045       314923.40     17.000   
4              FDY59               OUT018        64782.34      8.195   

      item_contenido_grasa  item_visibilidad            item_tipo  \
0  Bajo Contenido de Grasa          0.031230   Frutas y Vegetales   
1  Bajo Contenido de Grasa          0.042357  Bebidas Alcohólicas   
2  Bajo Contenido de Grasa          0.170500            Enlatados   
3                  Regular          0.026623            Enlatados   
4  Bajo Contenido de Grasa          0.000000             Dulcería   

   item_precio item_cantidad_venta  tienda_ano_establecimiento tienda_tamano  \
0       160.36                None                        2019         Medio   
1       123.24                None                        2009         Medio   
2       184.13                None                        2019         Medio   
3       144.25                None                        2012          None   
4        93.15                None                        2019         Medio   

  tienda_tipo_ubicacion          tienda_tipo  
0               Nivel 3  Supermercado Tipo 2  
1               Nivel 1  Supermercado Tipo 1  
2               Nivel 3  Supermercado Tipo 2  
3               Nivel 2  Supermercado Tipo 1  
4               Nivel 3  Supermercado Tipo 2

# **Desafio: ¡hora de practicar!**

Un colega del equipo de datos de Zoop solicitó el análisis de otro conjunto de datos además del enviado por la dirección. Estos datos se refieren a las ventas totales obtenidas por 896 franquicias en el primer mes del trimestre de análisis de Zoop.
Para acceder a los datos, puedes usar el siguiente código con el enlace abajo:

url_franquicias = 'https://gist.githubusercontent.com/ahcamachod/68eede222bdf1ba0ca413d529e800e7c/raw/576b0f3856467817df0fcf041d4a0787579917b0/ventas_franquicia.json'

Sabiendo esto, vamos al desafío:

Con la ayuda de ChatGPT, busca recolectar y acceder a los datos del archivo que están en formato JSON. Es importante dejar todos los datos de manera accesible, evitando estructuras que recuerden a un archivo JSON, como diccionarios.

Diccionario de datos: ventas_franquicia.json

Columna	          Descripción
ID_tienda	        Código identificador de las franquicias
ventas_por_dia	  Promedio del total obtenido en ventas por día
ventas_tienda	    Ventas totales obtenidas en un mes
tamano_tienda	    Categoría del total obtenido en ventas ('Pequeño' y 'Medio')
cliente_por_dia	  Promedio del total de clientes por día
items_disponibles	Número de artículos diferentes disponibles

En primer lugar hacemos un pedido a ChatGPT con el siguiente Prompt:
Prompt

Estamos actuando como científicos de datos en una empresa de supermercados.

Recibimos datos en formato de archivo JSON, estructurados en listas y diccionarios. El archivo está disponible en una URL de GitHub, que está guardada en una variable llamada "url_franquicias".

Programamos en Python y tenemos conocimiento de las bibliotecas de análisis de datos Numpy y Pandas. Por lo tanto, priorice el uso de estas bibliotecas.

Por favor, infórmenos cómo podemos leer estos datos en formato JSON y cómo podemos visualizarlos en formato de tabla en Python.

Con la respuesta lo aplicamos a nuestro problema:

import pandas as pd

import requests

url_franquicias = 'https://gist.githubusercontent.com/ahcamachod/68eede222bdf1ba0ca413d529e800e7c/raw/576b0f3856467817df0fcf041d4a0787579917b0/ventas_franquicia.json'

/* Descargar el JSON desde la URL

response = requests.get(url_franquicias)
datos_json = response.json()  /* Convertir la respuesta en un diccionario de Python

/* Crear un DataFrame de Pandas
df = pd.DataFrame(datos_json)

/* Mostrar las primeras filas
print(df.head())

/* en Jupyter Notebook usa:
df.head()

In [74]:
import pandas as pd
import numpy as np
import requests

url_franquicias = 'https://gist.githubusercontent.com/ahcamachod/68eede222bdf1ba0ca413d529e800e7c/raw/576b0f3856467817df0fcf041d4a0787579917b0/ventas_franquicia.json'

# Descargar el JSON desde la URL
response = requests.get(url_franquicias)
datos_json = response.json()  # Convertir la respuesta en un diccionario de Python

# Crear un DataFrame de Pandas
df_procesado = pd.DataFrame(datos_json)

# print(df)

# Mostrar las primeras filas
print(df_procesado.head())
df_procesado.head()

  ID_tienda                                             ventas  \
0    OUT431  {'ventas_por_dia': None, 'ventas_tienda': 8222...   
1    OUT738  {'ventas_por_dia': None, 'ventas_tienda': 1032...   
2    OUT426  {'ventas_por_dia': None, 'ventas_tienda': 5548...   
3    OUT948  {'ventas_por_dia': None, 'ventas_tienda': 9641...   
4    OUT139  {'ventas_por_dia': None, 'ventas_tienda': 5680...   

                                              tienda  
0  {'tamano_tienda': 'pequeno', 'cliente_por_dia'...  
1  {'tamano_tienda': 'medio', 'cliente_por_dia': ...  
2  {'tamano_tienda': 'pequeno', 'cliente_por_dia'...  
3  {'tamano_tienda': 'Pequeno', 'cliente_por_dia'...  
4  {'tamano_tienda': 'Pequeno', 'cliente_por_dia'...  


ID_tienda                                             ventas  \
0    OUT431  {'ventas_por_dia': None, 'ventas_tienda': 8222...   
1    OUT738  {'ventas_por_dia': None, 'ventas_tienda': 1032...   
2    OUT426  {'ventas_por_dia': None, 'ventas_tienda': 5548...   
3    OUT948  {'ventas_por_dia': None, 'ventas_tienda': 9641...   
4    OUT139  {'ventas_por_dia': None, 'ventas_tienda': 5680...   

                                              tienda  
0  {'tamano_tienda': 'pequeno', 'cliente_por_dia'...  
1  {'tamano_tienda': 'medio', 'cliente_por_dia': ...  
2  {'tamano_tienda': 'pequeno', 'cliente_por_dia'...  
3  {'tamano_tienda': 'Pequeno', 'cliente_por_dia'...  
4  {'tamano_tienda': 'Pequeno', 'cliente_por_dia'...

podemos solicitar nuevamente la ayuda de ChatGPT.

Prompt:

El DataFrame "df" tiene las columnas "ID_tienda", "ventas" y "tienda".

Las columnas "ventas" y "tienda" contienen diccionarios anidados dentro de cada una de las filas. ¿Cómo puedo transformar los datos de estos diccionarios y columnas para hacerlos más accesibles en mi DataFrame?

In [75]:
df_ventas = pd.json_normalize(df_procesado['ventas'])
df_tienda = pd.json_normalize(df_procesado['tienda'])
# Concatenar los DataFrames con los datos originales (excluyendo las columnas anidadas)
df_final = pd.concat([df_procesado.drop(columns=['ventas', 'tienda']), df_ventas, df_tienda], axis=1)
# Mostrar las primeras filas del DataFrame final
print(df_final.head())
df_final.head()

  ID_tienda  ventas_por_dia  ventas_tienda tamano_tienda  cliente_por_dia  \
0    OUT431             NaN   82220.053714       pequeno            330.0   
1    OUT738             NaN  103231.728424         medio            310.0   
2    OUT426             NaN   55487.996318       pequeno            870.0   
3    OUT948             NaN   96417.999686       Pequeno           1560.0   
4    OUT139             NaN   56809.025487       Pequeno             10.0   

   items_disponibles  
0               1968  
1               2189  
2               1736  
3               1099  
4               1516  


ID_tienda  ventas_por_dia  ventas_tienda tamano_tienda  cliente_por_dia  \
0    OUT431             NaN   82220.053714       pequeno            330.0   
1    OUT738             NaN  103231.728424         medio            310.0   
2    OUT426             NaN   55487.996318       pequeno            870.0   
3    OUT948             NaN   96417.999686       Pequeno           1560.0   
4    OUT139             NaN   56809.025487       Pequeno             10.0   

   items_disponibles  
0               1968  
1               2189  
2               1736  
3               1099  
4               1516

# **Clase 2 - Exploración de los datos**

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>

El DataFrame "df" contiene datos referentes a las ventas de diversos productos en varias tiendas del supermercado. Se proporcionan los IDs de productos y los IDs de tiendas que comercializaron esos productos, el total recaudado en ventas, así como descripciones sobre el producto vendido y la tienda que lo vendió.

Estamos iniciando la etapa de exploración de datos, en la cual deseamos conocer los valores informados e identificar si existen valores ausentes, duplicados y otras inconsistencias.

Construye un código para ejecutar la etapa de exploración de datos usando Python, comentando el paso a paso de manera clara y objetiva.

No es necesario utilizar "print()" para mostrar el resultado de las consultas.

In [9]:
# 1. Obtener una vista rápida de las primera filas del DataFrame, util para comprender la estructura general y verificar
# si los datos parecen correctos a primera vista
df.head()

item_identificador tienda_identificador  ventas_totales  item_peso  \
0              FDB08               OUT018       176503.58      6.055   
1              DRQ35               OUT049       185758.20      9.300   
2              FDD14               OUT018       165983.94     20.700   
3              FDY37               OUT045       314923.40     17.000   
4              FDY59               OUT018        64782.34      8.195   

      item_contenido_grasa  item_visibilidad            item_tipo  \
0  Bajo Contenido de Grasa          0.031230   Frutas y Vegetales   
1  Bajo Contenido de Grasa          0.042357  Bebidas Alcohólicas   
2  Bajo Contenido de Grasa          0.170500            Enlatados   
3                  Regular          0.026623            Enlatados   
4  Bajo Contenido de Grasa          0.000000             Dulcería   

   item_precio item_cantidad_venta  tienda_ano_establecimiento tienda_tamano  \
0       160.36                None                        2019         Medio   
1       123.24                None                        2009         Medio   
2       184.13                None                        2019         Medio   
3       144.25                None                        2012          None   
4        93.15                None                        2019         Medio   

  tienda_tipo_ubicacion          tienda_tipo  
0               Nivel 3  Supermercado Tipo 2  
1               Nivel 1  Supermercado Tipo 1  
2               Nivel 3  Supermercado Tipo 2  
3               Nivel 2  Supermercado Tipo 1  
4               Nivel 3  Supermercado Tipo 2

In [10]:
# 2. Calcular las estadísticas descriptivas para las columnas numéricas, nos dara una idea del rango de valores, medias y posibles outliers
df.describe()

ventas_totales    item_peso  item_visibilidad  item_precio  \
count    8.550000e+03  7081.000000       8550.000000  8550.000000   
mean     2.181949e+05    12.855023          0.066150   141.007453   
std      1.708098e+05     4.643508          0.051578    62.333062   
min      3.329000e+03     4.555000          0.000000    31.290000   
25%      8.349132e+04     8.770000          0.027024    93.787500   
50%      1.794331e+05    12.600000          0.053978   142.935000   
75%      3.100963e+05    16.850000          0.094646   185.760000   
max      1.308696e+06    21.350000          0.328391   266.890000   

       tienda_ano_establecimiento  
count                 8550.000000  
mean                  2007.830409  
std                      8.372541  
min                   1995.000000  
25%                   1997.000000  
50%                   2009.000000  
75%                   2014.000000  
max                   2019.000000

In [11]:
# 3. Verificar la estructura general y el tipo de datos en cada columna, esto ayuda a identificar tipos de datos incorrectos y columnas potenciales irrelevantes
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8550 entries, 0 to 8549
Data columns (total 13 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   item_identificador          8550 non-null   object 
 1   tienda_identificador        8550 non-null   object 
 2   ventas_totales              8550 non-null   float64
 3   item_peso                   7081 non-null   float64
 4   item_contenido_grasa        8550 non-null   object 
 5   item_visibilidad            8550 non-null   float64
 6   item_tipo                   8550 non-null   object 
 7   item_precio                 8550 non-null   float64
 8   item_cantidad_venta         0 non-null      object 
 9   tienda_ano_establecimiento  8550 non-null   int64  
 10  tienda_tamano               6133 non-null   object 
 11  tienda_tipo_ubicacion       8550 non-null   object 
 12  tienda_tipo                 8550 non-null   object 
dtypes: float64(4), int64(1), object(8

Observando esto último se nota que hay tres columnas que le faltan datos: item_peso, item_cantidad_venta y tienda_tamano


In [12]:
df.info(memory_usage='deep')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8550 entries, 0 to 8549
Data columns (total 13 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   item_identificador          8550 non-null   object 
 1   tienda_identificador        8550 non-null   object 
 2   ventas_totales              8550 non-null   float64
 3   item_peso                   7081 non-null   float64
 4   item_contenido_grasa        8550 non-null   object 
 5   item_visibilidad            8550 non-null   float64
 6   item_tipo                   8550 non-null   object 
 7   item_precio                 8550 non-null   float64
 8   item_cantidad_venta         0 non-null      object 
 9   tienda_ano_establecimiento  8550 non-null   int64  
 10  tienda_tamano               6133 non-null   object 
 11  tienda_tipo_ubicacion       8550 non-null   object 
 12  tienda_tipo                 8550 non-null   object 
dtypes: float64(4), int64(1), object(8

¡Realmente el uso de memoria es de 4.4MB!

In [13]:
# 4. Identificar valores ausentes en cada columna, esto ayuda a detectar columnas con datos inompletos y definir estrategias para manejar estos valores
df.isnull().sum()

item_identificador               0
tienda_identificador             0
ventas_totales                   0
item_peso                     1469
item_contenido_grasa             0
item_visibilidad                 0
item_tipo                        0
item_precio                      0
item_cantidad_venta           8550
tienda_ano_establecimiento       0
tienda_tamano                 2417
tienda_tipo_ubicacion            0
tienda_tipo                      0
dtype: int64

Veo los valores faltantes

In [14]:
# 5. Comprobar la cantidad de filas duplicadas, esto puede afectar los resultados del análisis, es importante detectarlos y decidir si elimnarlos o no.
df.duplicated().sum()

np.int64(27)

Tenemos 27 filas repetidas en mi DataFrame

In [15]:
# 6. Revisar la cantidad de valores únicos en columnas relevantes, esto permite identificar categóricas con una baja variedad de valores, lo cual
# indicaría errores o valores predeterminados incorrectos.
df.nunique()


item_identificador            1559
tienda_identificador            10
ventas_totales                3493
item_peso                      415
item_contenido_grasa             5
item_visibilidad              7880
item_tipo                       16
item_precio                   4839
item_cantidad_venta              0
tienda_ano_establecimiento       9
tienda_tamano                    3
tienda_tipo_ubicacion            3
tienda_tipo                      4
dtype: int64

Se detecta que hay un valor no unico diferente de 2 en item_contenido_grasa y en tienda_tipo	4

# **Desafio: ¡hora de practicar!**

Continuaremos los desafíos con otro conjunto de datos de Zoop.

En el último desafío, logramos acceder a los datos y tener acceso a todas las columnas de información. Ahora es el momento de poner en práctica la etapa de exploración de datos, enfocada en la comprensión de los datos y la búsqueda de inconsistencias.

Por lo tanto, vamos al desafío:

Con la ayuda de ChatGPT, realiza el estudio de los datos presentes en cada columna del conjunto de datos e identifica las inconsistencias que puedan estar presentes, como datos ausentes o filas duplicadas.

**Prompt**:
El DataFrame "df" contiene datos relativos a las ventas de varias franquicias dentro de un mes de operación. Se informan los valores totales obtenidos en ventas, así como características de la tienda, cantidad de clientes por día y artículos comercializados.

Estamos iniciando la etapa de exploración de datos, en la cual deseamos conocer los valores informados e identificar si existen valores ausentes, duplicados y otras inconsistencias.

Por favor, construye un código para ejecutar la etapa de exploración de datos usando Python.

No es necesario utilizar "print()" para mostrar el resultado de las consultas.

**Respuesta del ChatGPT:**
Aquí tienes un código en Python para explorar los datos en el DataFrame df usando Pandas. Este análisis nos permitirá detectar valores ausentes, duplicados y posibles inconsistencias.


In [ ]:
'''
import pandas as pd
import requests

# Descargar y cargar los datos en un DataFrame
url_franquicias = 'https://gist.githubusercontent.com/ahcamachod/68eede222bdf1ba0ca413d529e800e7c/raw/576b0f3856467817df0fcf041d4a0787579917b0/ventas_franquicia.json'
response = requests.get(url_franquicias)
datos_json = response.json()
df = pd.DataFrame(datos_json)

# Expandir los diccionarios dentro de 'ventas' y 'tienda'
df_ventas = pd.json_normalize(df['ventas']).add_prefix('ventas_')
df_tienda = pd.json_normalize(df['tienda']).add_prefix('tienda_')
df = pd.concat([df.drop(columns=['ventas', 'tienda']), df_ventas, df_tienda], axis=1)

# Resumen general del DataFrame
df.info()  # Muestra tipos de datos y valores nulos

# Estadísticas descriptivas de las columnas numéricas
df.describe()

# Verificar valores nulos en cada columna
df.isnull().sum()

# Identificar valores duplicados
df.duplicated().sum()

# Detectar valores únicos por columna (puede ayudar a encontrar inconsistencias)
df.nunique()

# Identificar registros con valores negativos en ventas o cantidad de clientes
df[(df.filter(like='ventas_') < 0).any(axis=1) | (df['ventas_clientes_dia'] < 0)]
'''

In [76]:
# 1. Verifica las primeras líneas del DataFrame
df_final.head()

ID_tienda  ventas_por_dia  ventas_tienda tamano_tienda  cliente_por_dia  \
0    OUT431             NaN   82220.053714       pequeno            330.0   
1    OUT738             NaN  103231.728424         medio            310.0   
2    OUT426             NaN   55487.996318       pequeno            870.0   
3    OUT948             NaN   96417.999686       Pequeno           1560.0   
4    OUT139             NaN   56809.025487       Pequeno             10.0   

   items_disponibles  
0               1968  
1               2189  
2               1736  
3               1099  
4               1516

Destacados:

La columna venta_por_dia podría tener valores nulos.
La columna tamano_tienda parece no tener datos estandarizados debido a la presencia de "pequeno", "Pequeno" y "Medio".

In [77]:
# 2. Resumen estadístico de las columnas numéricas
df_final.describe()

ventas_por_dia  ventas_tienda  cliente_por_dia  items_disponibles
count         1.00000     898.000000       880.000000         898.000000
mean       4294.64706   71071.894236       786.443182        1781.561247
std               NaN   22143.481947       267.038209         299.716498
min        4294.64706   19371.958539        10.000000         932.000000
25%        4294.64706   54510.922957       600.000000        1574.500000
50%        4294.64706   69228.128375       780.000000        1773.000000
75%        4294.64706   86119.754844       970.000000        1982.000000
max        4294.64706  152897.262005      1560.000000        2667.000000

Destacados:

Observamos que la columna venta_por_dia tiene solo 1 valor no nulo (4294.65).
En la columna cliente_por_dia hay una franquicia con solo 10 visitas por día, lo que podría ser relevante dependiendo del tamaño y características de la tienda.

In [78]:
# 3. Información general sobre el DataFrame
df_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 898 entries, 0 to 897
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   ID_tienda          898 non-null    object 
 1   ventas_por_dia     1 non-null      float64
 2   ventas_tienda      898 non-null    float64
 3   tamano_tienda      878 non-null    object 
 4   cliente_por_dia    880 non-null    float64
 5   items_disponibles  898 non-null    int64  
dtypes: float64(3), int64(1), object(2)
memory usage: 42.2+ KB


Destacados:

Se identificaron 898 filas y 6 columnas.
Las columnas venta_por_dia, tamano_tienda y cliente_por_dia tienen valores nulos.
La columna cliente_por_dia está definida como float64, pero los datos corresponden a valores enteros.

In [79]:
# 4. Verifica si existen valores duplicados
df_final.duplicated().sum()

np.int64(2)

Destacados:

Hay 2 filas duplicadas.

In [80]:
# 5. Verifica si existen valores ausentes por columna
df_final.isnull().sum()

ID_tienda              0
ventas_por_dia       897
ventas_tienda          0
tamano_tienda         20
cliente_por_dia       18
items_disponibles      0
dtype: int64

Destacados:

Confirmamos los valores nulos mencionados en el resultado anterior:
venta_por_dia tiene 897 valores nulos.
tamano_tienda tiene 20 valores nulos.
cliente_por_dia tiene 18 valores nulos.
Las otras columnas no tienen valores nulos.

In [81]:
# 6. Cuenta de valores únicos en cada columna
df_final.nunique()

ID_tienda            896
ventas_por_dia         1
ventas_tienda        896
tamano_tienda          3
cliente_por_dia      130
items_disponibles    616
dtype: int64

Destacados:

Existen solo 896 franquicias registradas (hay 2 filas duplicadas).
Se presentan 3 categorías en tamano_tienda, que ya identificamos en el head().

In [82]:
# 7. Categorías únicas de tamano_tienda
df_final['tamano_tienda'].unique()

array(['pequeno', 'medio', 'Pequeno', None], dtype=object)

Destacados:

Confirmamos las 3 categorías no estandarizadas en tamano_tienda ("pequeno", "Pequeno", "Medio").
También encontramos valores nulos.

Resumen de la exploración:

Al realizar este análisis, encontramos las siguientes inconsistencias que deben ser tratadas:

Filas duplicadas.
Valores ausentes.
Datos categóricos fuera de estándar.
Datos numéricos del tipo int definidos como float.
Este resumen nos ayudará a tener claridad sobre las áreas que requieren tratamiento en las próximas etapas del análisis.

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>

El DataFrame "df" contiene datos relacionados con las ventas durante un trimestre de un supermercado. Deseo utilizarlo para verificar el desempeño de esas ventas.

Durante la exploración del "df", identificamos algunas inconsistencias, como:

* Filas duplicadas,
* Valores nulos y
* Datos categóricos que fueron escritos de manera no estandarizada, dando la idea de que existen más categorías de las que realmente existen.

¿Qué debo hacer para garantizar que esas inconsistencias que encontré no afecten mi análisis?

# **Clase 3 - Limpieza y tratamiento de los datos**

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>


Durante la exploración del DataFrame "df", identificamos 27 filas duplicadas usando "df.duplicated().sum()".

El objetivo es analizar los datos en "df" para medir el desempeño de una empresa durante un trimestre de ventas.

¿Qué debemos hacer con los datos duplicados para garantizar que no haya impacto negativo en el análisis?

In [20]:
# Eliminar filas duplicadas
df = df.drop_duplicates()

In [21]:
df.duplicated().sum()

np.int64(0)

In [24]:
# Comprobamos el numero de filas y columnas despues de la limpieza
df.shape

(8523, 13)

In [25]:
df

item_identificador tienda_identificador  ventas_totales  item_peso  \
0                 FDB08               OUT018       176503.58      6.055   
1                 DRQ35               OUT049       185758.20      9.300   
2                 FDD14               OUT018       165983.94     20.700   
3                 FDY37               OUT045       314923.40     17.000   
4                 FDY59               OUT018        64782.34      8.195   
...                 ...                  ...             ...        ...   
8545              FDY08               OUT010        28096.76      9.395   
8546              FDC41               OUT017       130163.90     15.600   
8547              NCQ53               OUT045       614533.40     17.600   
8548              FDL46               OUT017       164985.24     20.350   
8549              NCN30               OUT046        96541.00     16.350   

         item_contenido_grasa  item_visibilidad             item_tipo  \
0     Bajo Contenido de Grasa          0.031230    Frutas y Vegetales   
1     Bajo Contenido de Grasa          0.042357   Bebidas Alcohólicas   
2     Bajo Contenido de Grasa          0.170500             Enlatados   
3                     Regular          0.026623             Enlatados   
4     Bajo Contenido de Grasa          0.000000              Dulcería   
...                       ...               ...                   ...   
8545                  Regular          0.286345    Frutas y Vegetales   
8546  Bajo Contenido de Grasa          0.117575  Alimentos Congelados   
8547  Bajo Contenido de Grasa          0.018944             Abarrotes   
8548  bajo contenido de grasa          0.054363                Snacks   
8549                      BCG          0.016993              Cereales   

      item_precio item_cantidad_venta  tienda_ano_establecimiento  \
0          160.36                None                        2019   
1          123.24                None                        2009   
2          184.13                None                        2019   
3          144.25                None                        2012   
4           93.15                None                        2019   
...           ...                 ...                         ...   
8545       139.18                None                        2008   
8546        75.67                None                        2017   
8547       237.36                None                        2012   
8548       117.95                None                        2017   
8549        95.74                None                        2007   

     tienda_tamano tienda_tipo_ubicacion          tienda_tipo  
0            Medio               Nivel 3  Supermercado Tipo 2  
1            Medio               Nivel 1  Supermercado Tipo 1  
2            Medio               Nivel 3  Supermercado Tipo 2  
3             None               Nivel 2  Supermercado Tipo 1  
4            Medio               Nivel 3  Supermercado Tipo 2  
...            ...                   ...                  ...  
8545          None               Nivel 3              Mercado  
8546          None               Nivel 2  Supermercado Tipo 1  
8547          None               Nivel 2  Supermercado Tipo 1  
8548          None               Nivel 2  Supermercado Tipo 1  
8549       Pequeño               Nivel 1  Supermercado Tipo 1  

[8523 rows x 13 columns]

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>


Identificamos tres columnas en "df" que presentan valores nulos:

* La columna "item_cantidad_venta" tiene 8550 filas con valores nulos.
* La columna "item_peso" tiene 1469 filas con valores nulos.
* La columna "tienda_tamano" tiene 2417 filas con valores nulos.

Hubo un gran número de ventas de artículos durante el trimestre, y no hay justificación para la ausencia de datos en la columna "item_cantidad_venta", lo que la convierte en una columna prescindible.

La columna "item_peso" contiene valores del tipo float, representando el peso en gramos de los productos.

La columna "tienda_tamano" tiene tres categorías de tamaño para supermercados.

¿Cómo debo manejar estos datos nulos para garantizar que mi análisis no se vea afectado?

In [26]:
# Eliminar la columna "item_cantidad_venta" por su falta de datos relevantes
df = df.drop(columns=["item_cantidad_venta"])

In [28]:
# Rellenar valores nulos en 'item_peso' con la media
df["item_peso"] = df["item_peso"].fillna(df["item_peso"].mean())

In [29]:
df["tienda_tamano"] = df["tienda_tamano"].fillna('Desconocido')

In [30]:
# Verificar si aún hay valores nulos
df.isnull().sum()

item_identificador            0
tienda_identificador          0
ventas_totales                0
item_peso                     0
item_contenido_grasa          0
item_visibilidad              0
item_tipo                     0
item_precio                   0
tienda_ano_establecimiento    0
tienda_tamano                 0
tienda_tipo_ubicacion         0
tienda_tipo                   0
dtype: int64

In [31]:
df['item_contenido_grasa'].value_counts().index

Index(['Bajo Contenido de Grasa', 'Regular', 'BCG', 'reg',
       'bajo contenido de grasa'],
      dtype='object', name='item_contenido_grasa')

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>

La columna "item_conteudo_grasa" del DataFrame "df" presenta 5 valores únicos dados por: 'Bajo Contenido de Grasa', 'Regular', 'BCG', 'reg' y 'bajo contenido de grasa'.

Los valores únicos de la columna "item_contenido_grasa" deberían ser solo 'Bajo Contenido de Grasa' y 'Regular', pero fueron insertados sin estandarización, lo que resultó en más categorías de las deseadas.

¿Cómo puedo ajustar los datos de la columna "item_contenido_grasa" para mantener solo las dos categorías estándar?

In [32]:
# Reemplazar los valores inconsistentes con las categorias estandar
diccionario = {
    "bajo contenido de grasa": "Bajo Contenido de Grasa",
    "BCG": "Bajo Contenido de Grasa",
    "reg": "Regular",
    "regular": "Regular"
}
df["item_contenido_grasa"] = df["item_contenido_grasa"].replace(diccionario)

In [33]:
df['item_contenido_grasa'].value_counts().index

Index(['Bajo Contenido de Grasa', 'Regular'], dtype='object', name='item_contenido_grasa')

In [34]:
df.head()

item_identificador tienda_identificador  ventas_totales  item_peso  \
0              FDB08               OUT018       176503.58      6.055   
1              DRQ35               OUT049       185758.20      9.300   
2              FDD14               OUT018       165983.94     20.700   
3              FDY37               OUT045       314923.40     17.000   
4              FDY59               OUT018        64782.34      8.195   

      item_contenido_grasa  item_visibilidad            item_tipo  \
0  Bajo Contenido de Grasa          0.031230   Frutas y Vegetales   
1  Bajo Contenido de Grasa          0.042357  Bebidas Alcohólicas   
2  Bajo Contenido de Grasa          0.170500            Enlatados   
3                  Regular          0.026623            Enlatados   
4  Bajo Contenido de Grasa          0.000000             Dulcería   

   item_precio  tienda_ano_establecimiento tienda_tamano  \
0       160.36                        2019         Medio   
1       123.24                        2009         Medio   
2       184.13                        2019         Medio   
3       144.25                        2012   Desconocido   
4        93.15                        2019         Medio   

  tienda_tipo_ubicacion          tienda_tipo  
0               Nivel 3  Supermercado Tipo 2  
1               Nivel 1  Supermercado Tipo 1  
2               Nivel 3  Supermercado Tipo 2  
3               Nivel 2  Supermercado Tipo 1  
4               Nivel 3  Supermercado Tipo 2

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8523 entries, 0 to 8549
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   item_identificador          8523 non-null   object 
 1   tienda_identificador        8523 non-null   object 
 2   ventas_totales              8523 non-null   float64
 3   item_peso                   8523 non-null   float64
 4   item_contenido_grasa        8523 non-null   object 
 5   item_visibilidad            8523 non-null   float64
 6   item_tipo                   8523 non-null   object 
 7   item_precio                 8523 non-null   float64
 8   tienda_ano_establecimiento  8523 non-null   int64  
 9   tienda_tamano               8523 non-null   object 
 10  tienda_tipo_ubicacion       8523 non-null   object 
 11  tienda_tipo                 8523 non-null   object 
dtypes: float64(4), int64(1), object(7)
memory usage: 1.1+ MB


# **Desafio: ¡hora de practicar!**

Ejecutar la limpieza y tratamiento de los datos con la ayuda de ChatGPT, buscando definir soluciones posibles que sean aplicables a los datos. No olvides siempre tener en cuenta el contexto en el que estamos inmersos.

Aquí tienes el texto traducido al español con los cambios de nombres de columnas que solicitaste:

En el último desafío encontramos 4 incoherencias:

Líneas duplicadas;
Valores ausentes;
Datos categóricos fuera del estándar; y
Datos numéricos del tipo int definidos como float.
Como ya hicimos en video, buscaremos con la ayuda de ChatGPT cómo podemos manejar cada una de ellas. Comenzaremos con las líneas duplicadas.

Como se mencionó en la presentación de los datos, este conjunto debería informar sobre las "ventas totales obtenidas por 896 franquicias en el primer mes del trimestre de análisis de Zoop". Por lo tanto, no deberían existir estas líneas duplicadas.

Informaremos esto a ChatGPT y solicitaremos una solución.

Prompt:

Durante la exploración del DataFrame "df", identificamos 2 líneas duplicadas usando df.duplicated().sum().

El DataFrame "df" debería contener solo información de ventas de 896 franquicias, pero dos de ellas aparecieron repetidas, sin justificación alguna.

¿Qué debemos hacer con estas líneas duplicadas para garantizar que no haya un impacto negativo en el análisis? Para manejar las líneas duplicadas en su DataFrame df, puede eliminarlas para asegurarse de que no tengan un impacto negativo en su análisis. Puede hacerlo utilizando el método drop_duplicates() de Pandas. Así es cómo eliminar las líneas duplicadas:

In [83]:
df_final

ID_tienda  ventas_por_dia  ventas_tienda tamano_tienda  cliente_por_dia  \
0      OUT431             NaN   82220.053714       pequeno            330.0   
1      OUT738             NaN  103231.728424         medio            310.0   
2      OUT426             NaN   55487.996318       pequeno            870.0   
3      OUT948             NaN   96417.999686       Pequeno           1560.0   
4      OUT139             NaN   56809.025487       Pequeno             10.0   
..        ...             ...            ...           ...              ...   
893    OUT206             NaN   87875.440154         medio            810.0   
894    OUT370             NaN   29410.047183       Pequeno            780.0   
895    OUT960             NaN   57362.867709       Pequeno            660.0   
896    OUT535             NaN   61585.737092       pequeno            990.0   
897    OUT202             NaN   95416.455607       Pequeno           1170.0   

     items_disponibles  
0                 1968  
1                 2189  
2                 1736  
3                 1099  
4                 1516  
..                 ...  
893               2031  
894               1439  
895               1510  
896               1836  
897               1500  

[898 rows x 6 columns]

In [84]:
# Elimine las líneas duplicadas del DataFrame
df_final = df_final.drop_duplicates()

# Verifique nuevamente si existen valores duplicados después de la eliminación
df_final.duplicated().sum()

np.int64(0)

**Prompt:**
El DataFrame "df" tiene 896 filas. Identificamos tres columnas en "df" que presentan valores nulos:

895 valores nulos en "ventas_por_dia";
20 valores nulos en "tamano_tienda"; y
18 valores nulos en "cliente_por_dia".
La única cantidad significativa de valores nulos se encontró en "ventas_por_dia". Solo se completó un dato en "ventas_por_dia" y no sabemos por qué los demás no fueron llenados. Lo que sabemos es que esta columna se volvió inútil para el análisis.

La columna "tamano_tienda" tiene dos categorías de tamaño para los supermercados.

La columna "cliente_por_dia" contiene valores del tipo float, que representan el promedio de clientes obtenidos por día en cada franquicia.

¿Cómo debo manejar estos datos nulos para garantizar que mi análisis no se vea afectado?

**Respuesta:**

Manejar los datos nulos es una parte importante de la preparación de datos para análisis. Presentaré algunas opciones para manejar los datos nulos en las columnas mencionadas:

Columna "ventas_por_dia": Como mencionaste que la mayoría de los valores en esta columna están ausentes y solo un dato fue completado, la columna podría considerarse inútil para el análisis. En ese caso, puede optar por eliminar toda la columna:

df = df.drop('ventas_por_dia', axis=1)

Columna "tamano_tienda": Si esta columna contiene información importante sobre el tamaño de las tiendas, puede optar por rellenar los valores nulos con un valor predeterminado, como "Desconocido" o "No Informado". Esto preservaría la columna para el análisis:

df['tamano_tienda'].fillna('Desconocido', inplace=True)

Si hay solo dos categorías posibles (por ejemplo, "Pequena" y "Grande"), puede reemplazar los valores nulos con el valor más común:

df['tamano_tienda'].fillna(df['tamano_tienda'].mode()[0], inplace=True)

Columna "cliente_por_dia": Para esta columna, puede rellenar los valores nulos con el promedio o la mediana de los valores existentes para no perder información valiosa:

media_cliente_por_dia = df['cliente_por_dia'].mean()
df['cliente_por_dia'].fillna(media_cliente_por_dia, inplace=True)

O si prefiere la mediana:

mediana_cliente_por_dia = df['cliente_por_dia'].median()
df['cliente_por_dia'].fillna(mediana_cliente_por_dia, inplace=True)

Después de tomar estas medidas, tendrá un DataFrame con los valores tratados para las columnas que contenían datos nulos. Asegúrese de que las decisiones tomadas se alineen con el objetivo de su análisis y que los valores completados no introduzcan un sesgo indebido en los resultados.

In [85]:
df_final = df_final.drop('ventas_por_dia', axis=1)

Como destacó ChatGPT, tenemos pocos valores de categorías en "tamano_tienda". Sin embargo, también no sabemos si el código de las franquicias corresponde a un tamaño que sea el más común. No podemos obtener esa información ahora, por lo que es adecuado reemplazar los valores nulos por "No registrado":

In [86]:
df_final['tamano_tienda'] = df_final['tamano_tienda'].fillna('No registrado')

La columna "cliente_por_dia" tiene solo 18 valores nulos, y un cambio pequeño no impacta mucho en las estadísticas finales. Además, no podemos encontrar los resultados reales de esta base.

Sabiendo esto, podemos reemplazar esos datos por el valor promedio, tal como sugirió la IA.

In [87]:
media_cliente_por_dia = df_final['cliente_por_dia'].mean()
df_final['cliente_por_dia'] = df_final['cliente_por_dia'].fillna(media_cliente_por_dia)

In [88]:
df_final.isnull().sum()

ID_tienda            0
ventas_tienda        0
tamano_tienda        0
cliente_por_dia      0
items_disponibles    0
dtype: int64

Pasemos ahora a los datos categóricos fuera del estándar. Nuevamente, contaremos con la ayuda de ChatGPT.

**Prompt:**

La columna "tamano_tienda" del DataFrame "df" presenta 3 valores únicos: 'pequeno', 'Medio' y 'Pequeno'.

Los valores únicos de la columna "tamano_tienda" deberían ser solo 'Medio' y 'Pequeño', pero se insertaron sin estandarización, lo que resultó en más categorías de las deseadas.

¿Cómo puedo ajustar los datos de la columna "tamano_tienda" para mantener solo las dos categorías estándar?

**Respuesta:**

Para ajustar los datos en la columna "tamano_tienda" del DataFrame df y mantener solo las dos categorías estándar ('Medio' y 'Pequeño'), puede utilizar el método replace de Pandas para reemplazar los valores inconsistentes por los valores estándar deseados.

Reemplace los valores inconsistentes en la columna "tamano_tienda"

df['tamano_tienda'].replace({'pequeno': 'Pequeño', 'Pequeno': 'Pequeño','medio':'Medio'}, inplace=True)

Ahora, verifique nuevamente los valores únicos en la columna "tamano_tienda"

df['tamano_tienda'].unique()


Este código usa el método replace para reemplazar los valores 'pequeno' y 'Pequeno' por 'Pequeño', manteniendo solo dos categorías estándar ('Medio' y 'Pequeño') en la columna "tamano_tienda". El argumento inplace=True aplica los cambios directamente al DataFrame.

Después de ejecutar este código, la columna "tamano_tienda" debe contener solo las categorías estándar deseadas, y los valores inconsistentes serán tratados. Puede verificar esto usando df['tamano_tienda'].unique() para confirmar que ahora solo tiene 'Medio' y 'Pequeño'.

Aplicamos esta solución y corregimos el problema.

In [89]:
df_final['tamano_tienda'].unique()

array(['pequeno', 'medio', 'Pequeno', 'No registrado'], dtype=object)

In [90]:
# Reemplace los valores inconsistentes en la columna "tamano_tienda"
df_final['tamano_tienda'] = df_final['tamano_tienda'].replace({'pequeno': 'Pequeño', 'Pequeno': 'Pequeño','medio':'Medio'})

# Ahora, verifique nuevamente los valores únicos en la columna "tamano_tienda"
df_final['tamano_tienda'].unique()

array(['Pequeño', 'Medio', 'No registrado'], dtype=object)

Pasemos al problema del tipo de valor.

**Prompt:**

La columna "cliente_por_dia" del DataFrame "df" contiene valores registrados como tipo "float64". Sin embargo, la columna "cliente_por_dia" solo contiene datos enteros.

¿Cómo puedo ajustar los datos de la columna "cliente_por_dia" al tipo correspondiente a sus datos?

**Respuesta:**

Para ajustar el tipo de datos de la columna "cliente_por_dia" del DataFrame df y que coincida con los datos, puedes usar el método astype de Pandas para convertir los valores al tipo entero (int). Así es como hacerlo:

Convierte la columna "cliente_por_dia" al tipo entero (int)

df['cliente_por_dia'] = df['cliente_por_dia'].astype(int)

Con esta línea de código, estarás convirtiendo los valores de la columna "cliente_por_dia" al tipo entero (int), que es apropiado ya que mencionaste que esta columna solo contiene datos enteros. Asegúrate de que los datos no contengan valores no enteros, ya que la conversión a int truncará los decimales y cualquier valor no entero se redondeará al número entero más cercano.

Después de ejecutar esta línea de código, la columna "cliente_por_dia" contendrá valores del tipo entero (int). Verifica esto usando df['cliente_por_dia'].dtype para confirmar que el tipo de datos se ha cambiado con éxito.

La solución es simple y consiste en cambiar el tipo de la columna a entero utilizando el método astype. Esto funciona muy bien para resolver nuestro problema.

In [93]:
df_final['cliente_por_dia'].dtype

dtype('float64')

In [94]:
# Convierte la columna "cliente_por_dia" al tipo entero (int)
df_final['cliente_por_dia'] = df_final['cliente_por_dia'].astype(int)

In [95]:
df_final.info()
df_final['cliente_por_dia'].dtype

<class 'pandas.core.frame.DataFrame'>
Index: 896 entries, 0 to 897
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   ID_tienda          896 non-null    object 
 1   ventas_tienda      896 non-null    float64
 2   tamano_tienda      896 non-null    object 
 3   cliente_por_dia    896 non-null    int64  
 4   items_disponibles  896 non-null    int64  
dtypes: float64(1), int64(2), object(2)
memory usage: 42.0+ KB


dtype('int64')

# **Clase 4 - Visualización de datos**

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>

Los datos del DataFrame "df" están limpios y tratados. Ahora, deseo iniciar la etapa de construcción de gráficos para crear visualizaciones que ofrezcan un buen soporte al análisis del desempeño de ventas de la empresa.

Las columnas disponibles en "df" son las siguientes:
- "item_identificador": Número único asignado a cada ítem
- "tienda_identificador": Nombre único de la tienda
- "ventas_totales": Valor obtenido en ventas totales
- "item_peso": Peso del ítem en gramos
- "item_contenido_grasa": Contenido de grasa del ítem
- "item_visibilidad": Valor de posicionamiento de cada ítem
- "item_tipo": Tipo de utilidad del ítem
- "item_precio": Precio unitario del ítem
- "tienda_ano_establecimiento": Año de Establecimiento de la tienda
- "tienda_tamano": Tamaño de la tienda
- "tienda_tipo_ubicacion": Tipo de ubicación de la tienda
- "tienda_tipo": Tipo de tienda

El objetivo es encontrar patrones o información que se destaquen cuando nos referimos al total obtenido en ventas especificado en la columna "ventas_totales".

Con base en la información proporcionada, ¿qué tipos de visualizaciones serían útiles para alcanzar el objetivo de analizar el desempeño de ventas?

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>

El DataFrame "df" contiene la columna "ventas_totales", que informa el valor en reales obtenido en ventas de productos por varias tiendas.

Además, en "df", tenemos la columna "tienda_tamano" con 4 categorías de tamaño para tiendas: 'Grande', 'Medio', 'Pequeño' y 'No registrado'.

Deseo crear una visualización que permita identificar el total obtenido en ventas para las cuatro categorías de tamaño de tienda y representarlas en un gráfico de barras.

Los datos en la columna "ventas_totales" necesitan ser agrupados por las categorías de "tienda_tamano".

La figura del gráfico debe seguir la identidad visual de la tienda, con el fondo de la figura y del gráfico en color "#ebebeb" y las barras en color "#e23155".

Además, es necesario mostrar el total de las ventas en millones encima de cada columna, con el texto de esta información en color "#19325c" y la estructura "MXN $ valor M".

Utiliza las bibliotecas de visualización de datos de Python para crear esta visualización.

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>


El DataFrame "df" contiene la columna "ventas_totales", que informa el valor en reales obtenido en ventas de productos por varias tiendas.

Además, en "df", tenemos la columna "tienda_tipo" con 4 categorías de tipo para tiendas: 'Supermercado Tipo 1', 'Supermercado Tipo 2', 'Mercado', 'Supermercado Tipo 3'.

Deseo crear una visualización que permita identificar el total obtenido en ventas para las cuatro categorías de tipo de tienda y representarlas en un gráfico de barras.

Los datos en la columna "ventas_totales" deben ser agrupados por las categorías de "tienda_tipo".

La figura de la gráfica debe seguir la identidad visual de la tienda, con el fondo de la figura y de la gráfica en color "#ebebeb" y las barras en color "#e23155".

Además, es necesario mostrar el total de las ventas en millones sobre cada columna, con el texto de esa información en color "#19325c" y la estructura "MXN$ valor M".

Utiliza las bibliotecas de visualización de datos de Python para crear esta visualización.

> #### <strong><font color="7b71ff" size =6>Prompt:</font></strong>

El DataFrame "df" contiene la columna "ventas_totales", que informa el valor en reales obtenido en ventas de productos por varias tiendas.

Además, en "df", tenemos la columna "item_tipo" con 16 categorías de productos.

Deseo crear una visualización que permita identificar el total obtenido en ventas por categorías de productos en un gráfico de barras horizontales.

Los datos en la columna "ventas_totales" deben agruparse por las categorías de "item_tipo".

La figura de la representación debe seguir la identidad visual de la tienda, con el fondo de la figura y del gráfico en el color "#ebebeb".

Las barras deben estar en el color "#e23155".

Los valores de las barras deben estar ordenados para mantener un aspecto de embudo.

Además, es necesario mostrar el total de las ventas en millones dentro o al lado de cada barra, con el texto de esa información en el color "#19325c" y en la estructura "MXN$ valor M".

No muestres la cuadrícula en el gráfico.

Utiliza las bibliotecas de visualización de datos de Python para crear esta visualización.